tests/run-pass/panic/good_unwind2.rs

```diff
@@ -0,0 +1,21 @@
+#![feature(c_unwind, unboxed_closures)]
+
+use std::panic;
+
+extern "C-unwind" fn good_unwind_c() {
+    panic!();
+}
+
+fn good_unwind_rust() {
+    panic!();
+}
+
+extern "rust-call" fn good_unwind_rust_call(_: ()) {
+    panic!();
+}
+
+fn main() {
+    panic::catch_unwind(|| good_unwind_c()).unwrap_err();
+    panic::catch_unwind(|| good_unwind_rust()).unwrap_err();
+    good_unwind_rust_call(());
+}
```

Comment:
This is a *lot* of new run-pass tests, can these be consolidated into larger tests testing more things? Our test suite will inevitably become slower over time, but there's no need to unnecessarily accelerate that process. ;)